File: client/app/src/main/java/com/birdthedeveloper/prometheus/android/prometheus/android/exporter/AndroidCustomExporter.java
package com.birdthedeveloper.prometheus.android.prometheus.android.exporter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import io.prometheus.client.Collector;
import io.prometheus.client.GaugeMetricFamily;

public class AndroidCustomExporter extends Collector {
    public List<MetricFamilySamples> collect() {
        List<MetricFamilySamples> mfs = new ArrayList<MetricFamilySamples>();
        // With no labels.
        mfs.add(new GaugeMetricFamily("my_gauge", "help", 42));
        // With labels
        GaugeMetricFamily labeledGauge = new GaugeMetricFamily("my_other_gauge", "help", Arrays.asList("labelname"));
        labeledGauge.addMetric(Arrays.asList("foo"), 4);
        labeledGauge.addMetric(Arrays.asList("bar"), 5);
        mfs.add(labeledGauge);
        return mfs;
    }
}

    // Registration
    //static final YourCustomCollector requests = new YourCustomCollector().register()

File: client/app/src/main/java/com/birdthedeveloper/prometheus/android/prometheus/android/exporter/AndroidCustomExporter.kt
package com.birdthedeveloper.prometheus.android.prometheus.android.exporter

import android.util.Log
import io.prometheus.client.Collector
import io.prometheus.client.Collector.MetricFamilySamples
import io.prometheus.client.GaugeMetricFamily
import kotlinx.coroutines.runBlocking
import java.util.Arrays
import kotlinx.coroutines.*

private val TAG = "ANDROID_EXPORTER"

class AndroidCustomExporter(metricEngine: MetricsEngine) : Collector() {
    private val metricsEngineRef = metricEngine

    override fun collect(): List<MetricFamilySamples> {
        Log.v(TAG, "collecting")
        val mfs: MutableList<MetricFamilySamples> = ArrayList()

        // metrics definitions
        collectBatteryStatus(mfs)

        return mfs
    }

    fun collectBatteryStatus(mfs: MutableList<MetricFamilySamples>) {
        //TODO
        val batteryPercentageGauge = GaugeMetricFamily(
            "battery_percentage", //TODO convert to ratio
            "Current battery percentage", listOf(),
        )

        val batteryPercentage : Double = metricsEngineRef.getBatteryPercentage().toDouble()
        batteryPercentageGauge.addMetric(listOf(), batteryPercentage)
        mfs.add(batteryPercentageGauge)
    }
}
